sanic/cookies/response.py

```diff
@@ -0,0 +1,619 @@
+from __future__ import annotations
+
+import re
+import string
+import sys
+
+from datetime import datetime
+from typing import TYPE_CHECKING, Any, Dict, List, Optional, Union, overload
+
+from sanic.exceptions import ServerError
+from sanic.log import deprecation
+
+
+if TYPE_CHECKING:
+    from sanic.compat import Header
+
+if sys.version_info < (3, 8):  # no cov
+    SameSite = str
+    LiteralRootPath = str
+    LiteralTrue = bool
+    LiteralFalse = bool
+else:  # no cov
+    from typing import Literal
+
+    LiteralRootPath = Literal["/"]
+    LiteralTrue = Literal[True]
+    LiteralFalse = Literal[False]
+    SameSite = Union[
+        Literal["Strict"],
+        Literal["Lax"],
+        Literal["None"],
+        Literal["strict"],
+        Literal["lax"],
+        Literal["none"],
+    ]
+
+DEFAULT_MAX_AGE = 0
+SAMESITE_VALUES = ("strict", "lax", "none")
+
+LEGAL_CHARS = string.ascii_letters + string.digits + "!#$%&'*+-.^_`|~:"
+UNESCAPED_CHARS = LEGAL_CHARS + " ()/<=>?@[]{}"
+TRANSLATOR = {
+    n: "\\%03o" % n for n in set(range(256)) - set(map(ord, UNESCAPED_CHARS))
+}
```

Comment:
Also meant to keep the improved version of this. This implementation from SimpleCookie would only work as intended if the string were Latin-1. It is not compatible with Werkzeug's implementation that operates on bytes instead, but the version I wrote (that only escapes ASCII) is compatible with that as well.

Reply:
added